wallet/src/account/mod.rs

```diff
@@ -254,10 +252,15 @@ impl Account {
         &self,
         db_tx: &impl WalletStorageReadUnlocked,
     ) -> WalletResult<(VRFPrivateKey, VRFPublicKey)> {
-        let vrf_key_path = make_path_to_vrf_key(&self.chain_config, self.account_index());
-        let private_key =
-            self.key_chain.get_private_key_for_path(&vrf_key_path, db_tx)?.private_key();
-        Ok(vrf_from_private_key(&private_key))
+        let (vrf_key_path, child_number) =
+            make_path_to_vrf_key(&self.chain_config, self.account_index());
+        let private_key = self.key_chain.get_private_key_for_path(&vrf_key_path, db_tx)?;
+        let (secret_key_bytes, _chain_code_bytes) =
+            private_key.to_hmac512_left_right_bytes(child_number);
```

Comment:
This function is still using secp256k1 key as intermediary. That has to go.

Reply:
it is the parent private key

Reply:
Still. There cannot be sec256k1 in the process. There seems to be a misunderstanding.

BIP-44 defines how we create a secret with length 512-bits. The left 256-bit should directly go into a `VRFPrivateKey::new_using_random_bytes()`. No secp256k1 should be involved. The secp256k1 is a temporary solution we made because we didn't have time to do the proper derivation.